lib/components/TextLinkButton/TextLinkButton.test.tsx

```diff
@@ -39,40 +39,40 @@ describe('TextLink', () => {
         </BraidTestProvider>,
       );
 
-      userEvent.tab();
-      userEvent.keyboard('{enter}');
+      await userEvent.tab();
+      await userEvent.keyboard('{enter}');
       expect(clickHandler).toHaveBeenCalledTimes(1);
       expect(clickHandler.mock.calls[0][0].type).toEqual('click');
       clickHandler.mockClear();
 
-      userEvent.keyboard('{space}');
+      await userEvent.keyboard(' ');
       expect(clickHandler).toHaveBeenCalledTimes(1);
       expect(clickHandler.mock.calls[0][0].type).toEqual('click');
       clickHandler.mockClear();
     });
 
-    it('should honour the tabIndex', () => {
+    it('should honour the tabIndex', async () => {
       const { getAllByRole } = render(
         <BraidTestProvider>
           <Text>
             <TextLinkButton onClick={() => {}} tabIndex={-1}>
               Not Focusable
             </TextLinkButton>
             <TextLinkButton onClick={() => {}} tabIndex={1}>
-              Focusable Second
+              Focusable First
             </TextLinkButton>
-            <TextLinkButton onClick={() => {}}>Focusable First</TextLinkButton>
+            <TextLinkButton onClick={() => {}}>Focusable Second</TextLinkButton>
```

Comment:
Test was wrong previously, according to [this issue](https://github.com/testing-library/user-event/issues/807) (and the browser 😄 ), focus will shift to tabIndex > 0 before tabIndex equal to 0.